lightning/src/ln/channel.rs

```diff
@@ -2832,13 +2857,35 @@ impl<Signer: Sign> Channel<Signer> {
 		}
 	}
 
+	/// Trigger the autoclose timer if it's in the starting position
+	pub fn maybe_trigger_autoclose_timer(&mut self, current_feerate: u32, new_feerate: u32, background_feerate: u32) -> bool {
```

Comment:
Why isn't this just a part of `send_update_fee`?

Reply:
There is at least a call in `update_channel_fee` to monitor stalling _inbound_ update_fee ?

Reply:
Right, but this method is *indirectly* called in timer tick (via update_channel_fee) but `check_channel_autoclose` is called *directly* in timer tick. I still don't see a reason why they are separate, they're both called during timer ticks.

Reply:
I've verified the paths. I see, you're thinking to just pass a boolean `trigger_autoclose` to `send_update_fee`  down from `update_channel_fee`. IIRC, my motivation to have a separate method to start the auto-close timer was to make the mechanism re-usable on conditions beyond feerate variations such as peer disconnection, fee-bumping reserve exhaustion, etc.